Assets/Scripts/Camera/PlayerCamera.cs

```diff
@@ -46,70 +53,69 @@ public void OnCamera(InputAction.CallbackContext context)
 
         public void OnSwitchView(InputAction.CallbackContext context)
         {
-            SwitchCameraPosition();
+            SwitchCameraView();
+        }
+
+        public void OnZoom(InputAction.CallbackContext context)
+        {
+            if (context.performed) isZoomed = true;
+            if (context.canceled) isZoomed = false;
         }
 
         private void Start()
         {
             Cursor.lockState = CursorLockMode.Locked;
             Cursor.visible = false;
             transform.position = cameraHolder.position;
-        }
 
-        private void Update()
-        {
-            AssignRotation();
-            RotateCamera();
+            cam = GetComponent<Camera>();
+            defaultFov = cam.fieldOfView;
+            zoomFov = defaultFov / zoomMagnification;
+
+            SwitchCameraView();
         }
 
-        private void AssignRotation()
+        private void Update()
         {
-            yRotation += xInput * Time.deltaTime * sensX;
-            xRotation -= yInput * Time.deltaTime * sensY;
-            xRotation = isFirstPerson
-                ? Mathf.Clamp(xRotation, rotationClampBottom, rotationClampTop)
-                : Mathf.Clamp(xRotation, rotationClampBottom - cameraChangeY / cameraChangeZ * 45f,
-                    rotationClampTop - cameraChangeY / cameraChangeZ * 45f);
+            HandleInput();
+            UpdateTransform();
+            UpdateFov();
         }
 
-        private void RotateCamera()
+        private void HandleInput()
         {
-            if (isFirstPerson)
-            {
-                transform.rotation = Quaternion.Euler(xRotation, yRotation, 0);
-            }
-            else
-            {
-                ChangePositionInThirdPerson();
-            }
+            Vector2 rotationDelta = new Vector2(yInput * Time.deltaTime * sensY, xInput * Time.deltaTime * sensX);
+            if (isZoomed) rotationDelta *= zoomSensitivityMultiplier;
+            yRotation += rotationDelta.y;
+            xRotation -= rotationDelta.x;
+            xRotation = Mathf.Clamp(xRotation, rotationClampBottom, rotationClampTop);
         }
 
-        private void ChangePositionInThirdPerson()
+        private void UpdateTransform()
         {
             var cameraRotationCenterPosition = cameraHolder.position;
-            var rayDirection = -transform.forward;
-            var cameraDistance = (float)Math.Sqrt(Math.Pow(cameraChangeZ, 2f) + Math.Pow(cameraChangeY, 2f));
+            var rayDirection = transform.rotation * cameraOffset;
+            var cameraDistance = cameraOffset.magnitude;
+            var mask = LayerMask.GetMask("Default");
```

Comment:
Extract class field:
```cs
public LayerMask cameraCollisionMask;
```
It's easier and safer to manage layers in inspector. `GetMask(string)` can easily fail if someone changes layer name in settings.